components/organisms/Chat/ChatMessages/CodeBlock.tsx

```diff
@@ -0,0 +1,93 @@
+import { Button } from "@/components/ui/Button";
+import { useContext, useEffect, useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+import { ChatContext } from "./ChatContext";
+import { SalesforceQueryResultWithError } from "@/shared/types/salesforceTypes";
+
+type CodeBlockProps = {
+  code: string;
+  setQueryResult: (arg: SalesforceQueryResultWithError) => void;
+  queryResult: SalesforceQueryResultWithError;
+};
+
+export const CodeBlock = ({
+  code,
+  setQueryResult,
+  queryResult,
+}: CodeBlockProps) => {
+  const [isLoadingQuery, setIsLoadingQuery] = useState(false);
+  const { append, isLoading } = useContext(ChatContext);
+
+  const runQuery = async () => {
+    setIsLoadingQuery(true);
```

Comment:
Why are there two loading states?

Reply:
One is returned by the useChat hook telling us when the chat is finished loading in. The other tells us when the query is finished running

Reply:
Why would a `CodeBlock` ever be rendered if the chat wasn't loaded in yet?

Reply:
the chat here isnt getting loaded in one go but bits and pieces . thus when there is text coming in, a new block gets rendered with the stream of text. 

Now, I render code blocks solely by if there is code text (starts and ends with '''). Sometimes i get text that gets enclosed in triple quotes, but still text is loading in. This led to some weird runtimes where it would run the code block throw an error and try to create a new message all while still creating the original message block.